### DISCUSSION TITLE
failed to config: requesting new api key for maps and places

### DISCUSSION THREAD

**Discussion by elfzhao** [photoprism/photoprism#2400] [General]
Hi,

I am a new user, after I installed the latest Photoprism Docker image, I could not access it. After enable debug log, I found out that seems the docker container cannot get new api key.

photoprism_1  | time="2022-06-04T10:56:37Z" level=info msg="PhotoPrism® needs your support!"
photoprism_1  | time="2022-06-04T10:56:37Z" level=info msg="Visit https://photoprism.app/membership to learn more."
photoprism_1  | time="2022-06-04T10:56:37Z" level=debug msg="config: running on 'Intel(R) Core(TM) i3-4010U CPU @ 1.70GHz', 3.5 GB memory detected"
photoprism_1  | time="2022-06-04T10:56:37Z" level=debug msg="settings: loaded from /photoprism/storage/config/settings.yml "
photoprism_1  | time="2022-06-04T10:56:37Z" level=debug msg="config: requesting new api key for maps and places"

photoprism_1  | time="2022-06-04T10:58:07Z" level=debug msg="config: Post \"https://hub.photoprism.app/v1/hello\": dial tcp 5.9.32.79:443: i/o timeout"

I can confirm the host can access Internet, the container is also connecting to the correct bridge, host enabled IP forwarding and iptables.

Any idea?

root@ihome-hub:/home/james/photoprism# docker network inspect photoprism_default
[
    {
        "Name": "photoprism_default",
        "Id": "72efe831574ddc39def81f91bbd1a057f8691460bc88664d93ef4311b6c2e24e",
        "Created": "2022-06-04T03:57:12.976501924Z",
        "Scope": "local",
        "Driver": "bridge",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": [
                {
                    "Subnet": "172.19.0.0/16",
                    "Gateway": "172.19.0.1"
                }
            ]
        },
        "Internal": false,
        "Attachable": true,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {
            "87a267f620427fba7f8ad70ef182d8ce3ca31a71fb0a093112e6419f26108271": {
                "Name": "photoprism_mariadb_1",
                "EndpointID": "3b5626eb909df1c449223927e6323d8e1ee5e06ac2b71ccc17c6bb33ac02a66f",
                "MacAddress": "02:42:ac:13:00:02",
                "IPv4Address": "172.19.0.2/16",
                "IPv6Address": ""
            },
            "9824c8004e49667ce4bb1ca106a44c7ce56647f1c4c8f3f96a82210d3609f9d3": {
                "Name": "photoprism_photoprism_1",
                "EndpointID": "bd550f4dab67289558bb4cd0c3f45c63e56c8e4167bd949d5c87b716f7ebd499",
                "MacAddress": "02:42:ac:13:00:03",
                "IPv4Address": "172.19.0.3/16",
                "IPv6Address": ""
            }
        },
        "Options": {},
        "Labels": {
            "com.docker.compose.network": "default",
            "com.docker.compose.project": "photoprism",
            "com.docker.compose.version": "1.29.2"
        }
    }
]
root@ihome-hub:/home/james/photoprism# sysctl net.ipv4.ip_forward
net.ipv4.ip_forward = 1
root@ihome-hub:/home/james/photoprism# cat /proc/sys/net/ipv4/ip_forward
1
root@ihome-hub:/home/james/photoprism# iptables -L
# Warning: iptables-legacy tables present, use iptables-legacy to see them
Chain INPUT (policy ACCEPT)
target     prot opt source               destination         

Chain FORWARD (policy DROP)
target     prot opt source               destination         
DOCKER-USER  all  --  anywhere             anywhere            
DOCKER-ISOLATION-STAGE-1  all  --  anywhere             anywhere            
ACCEPT     all  --  anywhere             anywhere             ctstate RELATED,ESTABLISHED
DOCKER     all  --  anywhere             anywhere            
ACCEPT     all  --  anywhere             anywhere            
ACCEPT     all  --  anywhere             anywhere            
ACCEPT     all  --  anywhere             anywhere             ctstate RELATED,ESTABLISHED
DOCKER     all  --  anywhere             anywhere            
ACCEPT     all  --  anywhere             anywhere            
ACCEPT     all  --  anywhere             anywhere            

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination         

Chain DOCKER (2 references)
target     prot opt source               destination         

Chain DOCKER-ISOLATION-STAGE-1 (1 references)
target     prot opt source               destination         
DOCKER-ISOLATION-STAGE-2  all  --  anywhere             anywhere            
DOCKER-ISOLATION-STAGE-2  all  --  anywhere             anywhere            
RETURN     all  --  anywhere             anywhere            

Chain DOCKER-ISOLATION-STAGE-2 (2 references)
target     prot opt source               destination         
DROP       all  --  anywhere             anywhere            
DROP       all  --  anywhere             anywhere            
RETURN     all  --  anywhere             anywhere            

Chain DOCKER-USER (1 references)
target     prot opt source               destination         
RETURN     all  --  anywhere             anywhere

**Comment by elfzhao** [comment]
sorted it out. just put the fix here in case someone encountered the same issue.

Seems the new Ubuntu 22.04 LTS has a restricted Iptables rules on forwarding table, after docker-compose start photoprism container, it did not create new iptables rule to allow forwarding traffic from inbound interface to new docker bridge interface. Need to manually add new "ufw route allow 2342/tcp" rule to enable inbound access to port 2342.

**Comment by enboig** [comment]
Running ubuntu 22.04, added "ufw route allow 2342/tcp":
```
# ufw status | grep 2342
2342                       ALLOW       Anywhere                  
2342 (v6)                  ALLOW       Anywhere (v6)             
2342/tcp                   ALLOW FWD   Anywhere                  
2342/tcp (v6)              ALLOW FWD   Anywhere (v6)  
```
Any hint?

**Reply by lastzero** [reply]
Note we've updated our troubleshooting checklists for you to more
explicitly mention firewall settings and include links to our other
documentation:

https://docs.photoprism.app/getting-started/troubleshooting/#maps-places
